realm/realm-library/src/objectServer/java/io/realm/InternalFunctions.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import org.bson.codecs.configuration.CodecRegistry;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.realm.internal.Util;
+import io.realm.internal.jni.JniBsonProtocol;
+import io.realm.internal.jni.OsJNIResultCallback;
+import io.realm.internal.network.ResultHandler;
+import io.realm.internal.objectstore.OsJavaNetworkTransport;
+import io.realm.log.RealmLog;
+import io.realm.mongodb.functions.Functions;
+
+
+/**
+ * Internal implementation of Functions invoking the actual OS function in the context of the
+ * {@link RealmUser}/{@link RealmApp}.
+ */
+class InternalFunctions extends Functions {
```

Comment:
I guess we should come up with a common name if we are to implement this pattern elsewhere.

Nitpick: Not sure how I feel about this compared to `FunctionsImpl`. Although `FunctionsImpl` is normally used when interfaces, it does feel more descriptive in my mind. But I don't have strong reasons and it is entirely subjective.

Reply:
Ok. Will go for `FunctionsImpl`